components/gfx/font_cache_thread.rs

```diff
@@ -74,40 +83,37 @@ impl FontTemplates {
     pub fn find_font_for_style(
         &mut self,
         desc: &FontTemplateDescriptor,
-    ) -> Option<Arc<FontTemplateData>> {
+    ) -> Option<FontTemplateRef> {
         // TODO(Issue #189): optimize lookup for
         // regular/bold/italic/bolditalic with fixed offsets and a
         // static decision table for fallback between these values.
         for template in &mut self.templates {
-            let maybe_template = template.data_for_descriptor(desc);
-            if maybe_template.is_some() {
-                return maybe_template;
+            if template.descriptor_matches(desc) {
+                return Some(template.clone());
             }
         }
 
         // We didn't find an exact match. Do more expensive fuzzy matching.
         // TODO(#190): Do a better job.
-        let (mut best_template_data, mut best_distance) = (None, f32::MAX);
-        for template in &mut self.templates {
-            if let Some((template_data, distance)) = template.data_for_approximate_descriptor(desc)
-            {
+        let (mut best_template, mut best_distance) = (None, f32::MAX);
+        for template in self.templates.iter() {
+            if let Some(distance) = template.descriptor_distance(desc) {
                 if distance < best_distance {
-                    best_template_data = Some(template_data);
+                    best_template = Some(template);
                     best_distance = distance
                 }
             }
         }
-        if best_template_data.is_some() {
-            return best_template_data;
+        if best_template.is_some() {
+            return best_template.cloned();
         }
 
         // If a request is made for a font family that exists,
         // pick the first valid font in the family if we failed
         // to find an exact match for the descriptor.
-        for template in &mut self.templates {
-            let maybe_template = template.get();
-            if maybe_template.is_some() {
-                return maybe_template;
+        for template in &mut self.templates.iter() {
+            if template.valid() {
```

Comment:
Do you think renaming this to `is_valid` would be more readable?

Reply:
Sure. That seems reasonable.